lib/mumukit/core/array.rb

```diff
@@ -12,4 +12,8 @@ def to_csv
       each { |hash| csv << hash.values }
     end
   end
+
+  def to_deep_struct
+    map { |val| (val.is_a?(Hash) || val.is_a?(Array)) ? val.to_deep_struct : val }
```

Comment:
It's kinda ugly and non-extensible to do those type-check. Also it is repeating code. I'd suggest to implement using `respond_to?` or even making object to understand `to_deep_struct` and answer `self` by default